### PR TITLE
feat(Pack - Xdsl) Adding page in case of missing RIO in invalid XDSL orders

### PR DESCRIPTION
**Pull request by feoche**
Accessible through `/pack/<PACK>/xdsl/<XDSL>/lines/<LINE>/missingRio`

### CODE REVIEW

File: client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.controller.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+angular.module("managerApp").controller("PackXdslMissingRioCtrl", function ($scope, $stateParams, $translate, $q, $timeout, $filter, OvhApiXdsl, Toast) {
```
Comment (antleblanc) [inline client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.controller.js]: -`$scope`

Comment (antleblanc) [inline client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.controller.js]: -`$scope`
-`$q`
-`$timeout`
-`$filter`

File: client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.html [hunk 2]
```diff
@@ -0,0 +1,91 @@
+<div class="telecom-pack-missing-rio">
+
+    <div class="telecom-section-content ml-0">
+
+        <oui-loader data-ng-if="PackXdslMissingRio.loading"
```
Comment (antleblanc) [inline client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.html]: ⚠️ must be `oui-spinner`.

File: client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.html [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                    data-size="m">
+        </oui-loader>
+
+        <div class="row" data-ng-if="!PackXdslMissingRio.loading">
```
Comment (antleblanc) [inline client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.html]: `.row > .col-xs-12` is just a single `div`.

File: client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.html [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                    <h2 class="widget-presentation-title" data-translate="xdsl_missing-rio_title"></h2>
+                </header>
+
+                <form name="missingRioForm"
+                      data-ng-submit="PackXdslMissingRio.sendRio()">
+                    <div class="row">
+                        <div class="col-md-6">
+                            <div class="form-group">
+
+                                <div class="form-group">
```
Comment (antleblanc) [inline client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.html]: nested `.form-group`?

File: client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.html [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                                 data-ng-class="{
+                                     'has-error': missingRioForm.rio.$touched && missingRioForm.rio.$invalid
+                                 }">
+                                <label for="rio"
+                                       class="control-label d-inline-block"
+                                       data-tooltip-placement="right"
+                                       data-tooltip-append-to-body="true"
+                                       data-uib-tooltip="{{ ::'xdsl_missing-rio_label_tooltip' | translate: {num: PackXdslMissingRio.number} }}">
+                                    <span data-translate="xdsl_missing-rio_label"></span>
+                                    <i class="ovh-font ovh-font-question ml-1"
```
Comment (antleblanc) [inline client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.html]: `Oui Icons` instead?

File: client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.html [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                                    <span data-ng-message="required"
+                                          data-translate="xdsl_missing-rio_required">
+                                    </span>
+                                </span>
+                            </div>
+                            <button
+                                type="submit"
+                                data-ng-disabled="missingRioForm.$invalid"
+                                class="btn btn-primary mt-2">
+                                <span data-translate="xdsl_missing-rio_send"></span>
```
Comment (antleblanc) [inline client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.html]: Apply `data-translate="xdsl_missing-rio_send"` directly on the `button` element.

File: client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.controller.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        self.missingRioForm.portNumber = true;
+
+        self.number = $stateParams.number;
+
+        return OvhApiXdsl.Lexi().get({
+            xdslId: $stateParams.serviceName
+        }).$promise.then(function (data) {
+            return data;
+        }, function () {
+            Toast.error($translate.instant("xdsl_missing-rio_init_error"));
```
Comment (Jisay) [inline client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.controller.js]: add the error message from API and reject the error :-)

File: client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.controller.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+            params.rio = self.missingRioForm.rio;
+        }
+
+        return OvhApiXdsl.Lexi().updateInvalidOrMissingRio({
+            xdslId: $stateParams.serviceName
+        }, params).$promise.then(function () {
+            Toast.success($translate.instant("xdsl_missing-rio_sent"));
+            self.init();
+        }, function () {
+            Toast.error($translate.instant("xdsl_missing-rio_error"));
```
Comment (Jisay) [inline client/app/telecom/pack/xdsl/missingRio/pack-xdsl-missing-rio.controller.js]: Same here : add the error message from API and reject the error :-)

**Pull request merged**